src/XProfile/VirtoCommerce.XProfile/Aggregates/MemberAggregateRootBase.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections.Generic;
+using System.Linq;
+using VirtoCommerce.CustomerModule.Core.Model;
+using VirtoCommerce.ExperienceApiModule.XProfile.Schemas;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.DynamicProperties;
+
+namespace VirtoCommerce.ExperienceApiModule.XProfile.Aggregates
+{
+    public interface IMemberAggregateRoot
+    {
+        public Member Member { get; set; }
+
+        MemberAggregateRootBase UpdateAddresses(IList<Address> addresses);
+        MemberAggregateRootBase UpdateDynamicProperties(IList<DynamicPropertyValue> values, IDynamicPropertyMetaDataResolver metaDataResolver, IDynamicPropertyDictionaryItemsSearchService dynamicPropertyDictionaryItemsSearchService);
+    }
+
+
+    public abstract class MemberAggregateRootBase : IMemberAggregateRoot
+    {
+        public virtual Member Member { get; set; }
+
+        public virtual MemberAggregateRootBase UpdateAddresses(IList<Address> addresses)
+        {
+            Member.Addresses = addresses.ToList();
+
+            return this;
+        }
+
+        public virtual MemberAggregateRootBase UpdateDynamicProperties(IList<DynamicPropertyValue> values, IDynamicPropertyMetaDataResolver metaDataResolver, IDynamicPropertyDictionaryItemsSearchService dynamicPropertyDictionaryItemsSearchService)
+        {
+            var memberDynamicProperties = Member.DynamicProperties.ToList();
+
+            foreach (var propValueGroup in values.GroupBy(x => x.Name))
+            {
+                var propertyValue = memberDynamicProperties.FirstOrDefault(x => x.Name.EqualsInvariant(propValueGroup.Key));

```

Comment:
All this code needs to be rewritten. The best solution will be to build the two DynamicObjectProperty collections source and target. The source collection is made from a member.DynamicProperties, the target - from input values, and finally a source collection must be merged to target by singe call (Patch).

Please get inspiration from there 
https://github.com/VirtoCommerce/vc-module-catalog/blob/dev/src/VirtoCommerce.CatalogModule.Data/Model/CatalogEntity.cs#L139

Reply:
done with the ability to update specified dynamic properties only (partial update)